snippets/csharp/buffers/MyClass.cs

```diff
@@ -0,0 +1,305 @@
+            }
+            else
+            {
+                // There are 4 bytes split across multiple segments. Since it's so small, it 
+                // can be copied to a stack allocated buffer. This avoids a heap allocation.
+                Span<byte> stackBuffer = stackalloc byte[4];
+                lengthSlice.CopyTo(stackBuffer);
+                length = BinaryPrimitives.ReadInt32BigEndian(stackBuffer);
+            }
+
+            // Move the buffer 4 bytes ahead.
+            buffer = buffer.Slice(lengthSlice.End);
+
+            return true;
+        }
+        #endregion
+
+        #region snippet6
+        static bool TryParseLine(ref ReadOnlySequence<byte> buffer, out ReadOnlySequence<byte> line)
+        {
+            SequencePosition position = buffer.Start;
+            SequencePosition previous = position;
+            var index = -1;
+            line = default;
+
+            while (buffer.TryGet(ref position, out ReadOnlyMemory<byte> segment))
+            {
+                ReadOnlySpan<byte> span = segment.Span;
+
+                // Look for \r in the current segment.
+                index = span.IndexOf((byte)'\r');
+
+                if (index != -1)
+                {
+                    // Check next segment for \n.
+                    if (index + 1 >= span.Length)
+                    {
+                        var next = position;
+                        if (!buffer.TryGet(ref next, out ReadOnlyMemory<byte> nextSegment))
+                        {
+                            // You're at the end of the sequence.
+                            return false;
+                        }
+                        else if (nextSegment.Span[0] == (byte)'\n')
+                        {
+                            //  A match was found.
+                            break;
+                        }
+                    }
+                    // Check the current segment of \n.
+                    else if (span[index + 1] == (byte)'\n')
+                    {
+                        // It was found.
+                        break;
+                    }
+                }
+
+                previous = position;
+            }
+
+            if (index != -1)
+            {
+                // Get the position just before the \r\n.
+                var delimeter = buffer.GetPosition(index, previous);
+
+                // Slice the line (excluding \r\n).
+                line = buffer.Slice(buffer.Start, delimeter);
+
+                // Slice the buffer to get the remaining data after the line.
+                buffer = buffer.Slice(buffer.GetPosition(2, delimeter));
+                return true;
+            }
+
+            return false;
+        }
+        #endregion
+
+        #region snippet7
+        static void EmptySegments()
+        {
+            // This logic creates a ReadOnlySequence<byte> with 4 segments but with a length 
```

Comment:
This sentence seems incomplete. Any suggestions here @davidfowl @Rick-Anderson?

Reply:
@halter73  can you suggest better wording?

Reply:
It seems complete to me. Maybe you're missing the "// of 2 of those segments are empty." on the next line.

It might read slightly better if the whole sentence read as follows instead:

```suggestion
            // This logic creates a ReadOnlySequence<byte> with 4 segments, two of which are empty.
```

If you apply this suggestion, make sure to delete the following line.

Reply:
I'll probably break the sentence in two to not create a super long comment and make scrollbars appear, but thanks!

Reply:
It was originally a one line long comment. I like @halter73  rewrite, much better.

Reply:
Me two. I just broke his suggestion in two lines.